core/ir/opnd_api.h

```diff
@@ -2234,13 +2243,28 @@ DR_API
  * Or a base register plus an optionally extended and shifted index register:
  * - [base_reg, index_reg, extend_type, shift_amount]
  *
- * The shift_amount is zero or, if \p scaled, a value determined by the
- * size of the operand.
+ * If \p scaled is enabled, \p shift determines the shift amount.
  *
  * The resulting operand has data size \p size (must be an OPSZ_ constant).
  * Both \p base_reg and \p index_reg must be DR_REG_ constants.
  * Either \p index_reg must be #DR_REG_NULL or disp must be 0.
  *
+ * TODO i#3044: WARNING this function may change during SVE development of
+ * DynamoRIO. The function will be considered stable when this warning has been
+ * removed.
```

Comment:
Hi @derekbruening  and @abhinav92003, this PR is the first of many SVE memory instructions we will be upstreaming in the coming weeks. There will be changes to the API which we plan to keep to a minimum. These API changes and additions may be to some extent fluid through February. We'll add a warning message like this when it is the case. We hope to have these changes stable by March. We will try and limit any unstable changes to AArch64 parts of the API. There will be no CI test regressions.